### DISCUSSION TITLE
Web Scraping the issues from Personal Repository for GitHub and GitHub Enterprise

### DISCUSSION THREAD

**Discussion by adrdediu** [community/community#56350] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Greetings Sir/Madam

I am writing this email as I came across an idea, I would like to save the content of all the issues that are posted on my repositories, and then make a database out of them so I could look through them. 

Is it mandatory to use the GitHub API in order to fetch all of the issue contents or would it be possible to save the HTML content for each page with a certain delay between each saving so as not to overload the servers with too many requests?

Also, I would like to know the answer to the question above in the case of GitHub Enterprise as well.

Thank you for your time and consideration and I look forward to hearing from you.

Yours Sincerely,
Adrian-Nicolae Dediu

**Comment by aahouzi** [comment, marked as answer]
To retrieve the content of issues from your repositories, using the GitHub API is the recommended and preferred method. The GitHub API provides a reliable and efficient way to programmatically interact with GitHub repositories, including accessing issue data.

By using the GitHub API, you can make requests to fetch issue information, including their titles, bodies, comments, labels, and other relevant data. This allows you to extract the necessary content and store it in a database for later retrieval and analysis.

Using the GitHub API also ensures that you stay within the permitted rate limits set by GitHub, preventing excessive requests that could overload the servers. The rate limits help maintain the performance and availability of the GitHub platform.

Regarding your question about saving the HTML content for each page with a delay between each saving, it's important to note that GitHub's Terms of Service prohibit the scraping or downloading of large amounts of data from their website. This includes saving HTML content directly from the GitHub website for multiple pages or repositories.

Instead, it is recommended to use the GitHub API to fetch the issue contents in a controlled and authorized manner. This way, you can ensure that you adhere to the terms and conditions set by GitHub.

As for GitHub Enterprise, the availability and functionality of the GitHub API might differ depending on the specific instance and configuration of the GitHub Enterprise version you are using. It's advisable to consult the GitHub Enterprise documentation or reach out to the GitHub Enterprise support team for more specific details on API access and usage within your particular setup.

I hope this information is helpful to you, and I encourage you to explore the GitHub API documentation for further details on accessing and retrieving issue data programmatically. If you have any further questions or require additional assistance, please feel free to ask.

**Comment by Charlotte-br560** [comment]
Hi Adrian-Nicolae,

It's recommended to use the GitHub API to scrape GitHub issues from your repositories. This ensures reliable and efficient access to the data while respecting GitHub's rate limits. Saving HTML content with delays might not be as reliable and can lead to issues with data consistency.

The same principles apply for GitHub Enterprise, but ensure you have appropriate access permissions.
